BlueprintUI/Tests/EnvironmentTests.swift

```diff
@@ -163,6 +181,47 @@ class EnvironmentTests: XCTestCase {
     }
 }
 
+
+class Environment_UIView_Tests : XCTestCase {
+    
+    func test_inheritedBlueprintEnvironment() {
+        
+        var environment = Environment.empty
+        environment[TestingKey1.self] = 1
+        
+        let first = View(subview: View(subview: View(subview: nil)))
+        let second = first.subviews[0]
+        let third = second.subviews[0]
+        
+        first.nativeViewNodeBlueprintEnvironment = environment
+        
+        XCTAssertEqual(first.inheritedBlueprintEnvironment?[TestingKey1.self], 1)
+        XCTAssertEqual(second.inheritedBlueprintEnvironment?[TestingKey1.self], 1)
+        XCTAssertEqual(third.inheritedBlueprintEnvironment?[TestingKey1.self], 1)
+        
+        first.nativeViewNodeBlueprintEnvironment = nil
+        
+        XCTAssertEqual(first.inheritedBlueprintEnvironment?[TestingKey1.self], nil)
+        XCTAssertEqual(second.inheritedBlueprintEnvironment?[TestingKey1.self], nil)
+        XCTAssertEqual(third.inheritedBlueprintEnvironment?[TestingKey1.self], nil)
```

Comment:
Should we also test setting a non-nil value on a non-root view and ensure that superviews are not resolving to a value?
```suggestion
        XCTAssertEqual(third.inheritedBlueprintEnvironment?[TestingKey1.self], nil)
        
        second.nativeViewNodeBlueprintEnvironment = 2

        XCTAssertEqual(first.inheritedBlueprintEnvironment?[TestingKey1.self], nil)
        XCTAssertEqual(second.inheritedBlueprintEnvironment?[TestingKey1.self], 2)
        XCTAssertEqual(third.inheritedBlueprintEnvironment?[TestingKey1.self], 2)
        
```

Reply:
I'm not quite sure I follow what you mean here – can you expand a bit?

Reply:
(If I'm reading the comment correctly; the concern I think is a superview finding a value from a subview? As implemented; that can't happen, since we only traverse up.)

Reply:
Yeah, I don't see how the test I'm suggesting would fail—no big deal if you want to omit it. It's really just asserting that you're not using a reference to the same environment in each layer of the tree.